sql/src/main/java/io/crate/jobs/PageDownstreamContext.java

```diff
@@ -56,6 +56,7 @@
     private final Streamer<?>[] streamers;
     private final RamAccountingContext ramAccountingContext;
     private final int numBuckets;
+    private final Set<Integer> buckets;
```

Comment:
maybe use IntHashSet to also avoid some boxing/unboxing

Reply:
We avoid some but we still keep others (see `fetchMore`). Performance-wise, I don't think the unboxing here is relevant at all.

Reply:
yep just saw that later on we cannot avoid it anyway at least with existing code since we have Integer all over and we treat also nulls.